Docs/PinterestFocusedXcodeProjects.md

```diff
@@ -0,0 +1,121 @@
+    script = "tools/instrumentation_helpers/log_build_start.py",
+)
+```
+
+Longer term we realized this wouldn't scale for the large number of diverse
+testing types. Xcode target settings are composed onto the build graph and
+specified adjacent to the Xcode target, mainly through custom macros. XCHammer
+aggregates all Xcode configuration for included targets [via an
+aspect](https://github.com/pinterest/xchammer/pull/192) which makes Xcode
+configuration composeable. By convention, the DSL is declared via macros
+adjacent to related BUILD files, e.g. for an an app declared at
+`Pinterest/iOS/App`, the Xcode configuration is declared at
+`Pinterest/iOS/App/XcodeProjectConfig.bzl`. The main BUILD file imports all of
+the configuration macros.
+
+### "XCFocus" aka focused projects
+
+Pinterest has a lot of source files which makes building, project generation,
+and indexing slow. We use Bazel locally to maximize caching and paralleism of
+clang invocations.
+
+Within the `XCHammerConfig` directory, users declare Xcode projects via the `xcode_project` rule in a way that's ignored from SCM. By default we focus by target and path to succinctly filter a subset of # transitive dependencies for specified targets.
+
+```
+# XCHammerConfig/Focux.bzl
+ pinterest_xcode_project(
+        targets = [ ":PinterestDevelopment" ],
+        paths = [ "Pinterest/**" ],
+    )
+```
+
+After editing the focused file, the Xcode project automatically is updated (
+add the this for Bazel builds )
+
+Generation time for focused are an order of magnitude faster the generating our entire Xcode workspace.
+
+
+### Indexing
+
+Like instrumentation, we import indexes into the Xcode project via scheme
+actions. After a build, an sync task is triggered that index-imports the index
+into Xcode.
+
+As XCHammer Xcode projects are designed to reproduce Bazel build via Xcode
+build, incremental indexing works as we type. Xcode picks up compiler settings
+from Xcode build targets.
+
+### Xcode testing Bazel built test bundles
+
+Xcode has deep integration into the test running system which adds complexity
+into how Bazel targets are integrated into the Project. In the original
+implementation of Xcode testing Bazel builds, the test bundle and optionally
+application are built with Bazel. Xcode is given a `TEST_HOST` environment
+variable pointing at the Bazel built application, which allows Xcode to setup
+the test and app.
+
+In order for "Focused" tests to run e.g. running a single test via the dot on
+the side, Xcode needs a reference to each file under test within the project.
+This adds complexity around indexing, and compiling, as we compile with Bazel.
+In order to pass XCBuild's output checking, we provide compilers and linkers
+that produce dummy output and exit 0. See the following section for more
+information.
+
+### Bazel integration
+
+XCHammer leverages both Tulsi's aspect code and build script. It simply creates "Bazel buildable" target that can run Bazel.
```

Comment:
- "build script" -> "run script phase"
- "It creates buildable targets that use Bazel as the build system"